tools/shared.py

```diff
@@ -834,20 +835,23 @@ def __init__(self, environ=os.environ):
       except Exception as e:
         logging.error(str(e) + 'Could not create canonical temp dir. Check definition of TEMP_DIR in ' + hint_config_file_location())
 
+  def apply(self):
+    global DEBUG, EMSCRIPTEN_TEMP_DIR, DEBUG_CACHE, CANONICAL_TEMP_DIR, TEMP_DIR
+    DEBUG = self.DEBUG
+    EMSCRIPTEN_TEMP_DIR = self.EMSCRIPTEN_TEMP_DIR
+    DEBUG_CACHE = self.DEBUG_CACHE
+    CANONICAL_TEMP_DIR = self.CANONICAL_TEMP_DIR
+    TEMP_DIR = self.TEMP_DIR
+
   def get_temp_files(self):
-    return tempfiles.TempFiles(
-      tmp=self.TEMP_DIR if not self.DEBUG else get_emscripten_temp_dir(),
-      save_debug_files=os.environ.get('EMCC_DEBUG_SAVE'))
-
-def apply_configuration():
-  global configuration, DEBUG, EMSCRIPTEN_TEMP_DIR, DEBUG_CACHE, CANONICAL_TEMP_DIR, TEMP_DIR
```

Comment:
So I don't see a `configuration` declared anywhere else in `shared`.
I didn't know `global` worked that way. That's kinda spooky.